### PR TITLE
Api test developement - refactor, move requests lib call to one module

### PR DESCRIPTION
**Pull request by kamrenata**

### CODE REVIEW

File: lib/client_requests.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+from faker import Faker
+from lib.payload_generator import StorePayload, UserPayload, UserCredentials
+
+
+class Pets(HTTPClient):
+    fake = Faker()
+
+    def __init__(self):
+        super().__init__()
+        # при наследовании, инициализация child & parent class
```
Comment (dimaklepikov) [inline lib/client_requests.py]: на самом деле, мы инициализируем наследуемый класс в этом, чтобы использовать его методы внутри текущего класса :)

Просто  FYI

File: lib/client_requests.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        return self.delete(route, headers=headers, params=None)
+
+
+class Users(HTTPClient):
+    user_payload = UserPayload()
+    user_credentials = UserCredentials()
+
+    def create_user_with_list(self):
+        route = "user/createWithList"
+        required_payload = self.user_payload.generate_users_payload_array()
```
Comment (dimaklepikov) [inline lib/client_requests.py:74]: Perfect implementation, where have you learned of it? :)

File: lib/payload_generator.py [hunk 3]
```diff
@@ -51,7 +51,24 @@ def generate_users_payload_multiple_array(self, user_amount):
...
+class StorePayload:
+    def __init__(self, pet):
+        self.pet = pet
+
+    def generate_place_pet_store_payload(self, true=None):
+        required_payload = {
+                "id": random.randint(1, 10),
+                "petId": self.pet.add_pet_and_get_id(),
+                "quantity": random.randint(1, 5),
+                "shipDate": "2024-09-10T11:57:49.135Z", # hardcoded date?
```
Comment (dimaklepikov) [inline lib/payload_generator.py]: One more point to improve:

Please, generate the following date format using `datetime` library:
https://docs.python.org/3/library/datetime.html

File: tests/api/test_pets.py [hunk 4]
```diff
@@ -1,27 +1,27 @@
...
         )
         response = response.json()
         for x in response:
             assert x["status"] == status
 
-    def test_get_by_id(self, add_and_delete_new_pet):
-        pet_id = add_and_delete_new_pet.json().get("id")
-        response = requests.get(f"{URL}/pet/{pet_id}")
+    def test_get_by_id(self):
+        # response = requests.get(f"{URL}/pet/{pet_id}")
```
Comment (dimaklepikov) [inline tests/api/test_pets.py]: Remove comment

File: tests/ui/test_base.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import pytest
+from pages.base_page import BasePage
+
+
+@pytest.mark.usefixtures("driver_for_base_test")
+class TestBase:
+
+    # def __init__(self, driver):
```
Comment (dimaklepikov) [inline tests/ui/test_base.py]: Comments :)

File: tests/ui/test_base.py
Comment (dimaklepikov) [inline tests/ui/test_base.py:1]: Is this file used in example/debugging purposes? If so, can be removed

**Review by dimaklepikov — CHANGES_REQUESTED** [review]
Perfect, looks more consistent and well-designed! Great work so far

Just manage to fix small comments & date generation proposal!

File: lib/client_requests.py (hunk 2 above)
Comment (kamrenata) [reply lib/client_requests.py:74]: From you or google I guess:)

File: lib/payload_generator.py (hunk 3 above)
Comment (kamrenata) [reply lib/payload_generator.py]: done

**Review by dimaklepikov — APPROVED** [review]

**Pull request merged**